### ISSUE TITLE
[Request] Request to add Quick Menu Closing Option

### ISSUE THREAD

**Issue by lucky80** [ssborbis/ContextSearch-web-ext#509]
Hello, I'm using this add-on very usefully.
Thank you.

I have a request, so I'm leaving this message.

After searching for a specific word in a search engine, block it, modify it to another word, and then press Enter to search.

However, when I pressed Enter for search, the quick menu that was shown disappears, and I have to press Enter one more time to search.

In other words, it's uncomfortable to press Enter twice.

Therefore, I would like to request the addition of the menu 'if keyboard input' to the closing option of the Quick Menu.

Also, it would be useful if there is an option that automatically closes after the set time after the quick menu opens.

Once again, thank you for the good program.

![Cap 2022-07-19 16-36-13-071](https://user-images.githubusercontent.com/1419119/179693692-d922b0fc-b4dd-4d98-b59a-ee29e475c56c.png)
.

**Comment by 7peanuts** [comment]
Hmm… which part needs to be searched by pressing Enter?

If you want to keep the menu open, you can set these two options and use `Esc` or mouse wheel to close it.

![20220719180202](https://user-images.githubusercontent.com/82173614/179724727-960daec0-99cc-437c-8bb2-7404510dc5cd.png)

**Comment by ssborbis** [comment]
What method are you using to open the quick menu? Auto? 

Is the problem that, when you select text in the searchbar on a webpage, the qm opens and you have to press Enter to get rid of it?

If so, you *could*:

1. disable the qm from opening on input elements ( CS Options -> Quick Menu -> Enable On -> uncheck input
2. hold a modifier key down before you release the left mouse button while selecting ( alt, ctrl, shift, whatever ), and the menu shouldn't open.

If it's something else, please explain a bit more. If you're able to make a GIF and post, that would also be helpful. There are so many ways this addon can be configured, I may not be seeing the same things you are.

I'm confused about what you said here:

> when I pressed Enter for search, the quick menu that was shown disappears

For me, pressing Enter initiates a search with the first engine in the list when the quick menu is open. Is that not happening in your case? ESC is the key I use to close the menu.

**Comment by lucky80** [comment]
First of all, thank you for your reply.

I'll write it in more detail.

After I search in the search text box, I want to change the text on the same site result screen.

Usually, when I want to search again, I double-click the word I searched for, specify the block by dragging, specify the block, and then enter another text.

With the above block designation, the Quick Menu will continue to pop up when I write the word to search, and when I type in the text and press the Enter key to search, the Quick Menu will disappear. And what I'm trying to do is I'm trying to search for the words that I typed, and even though I typed the enter key, only the quick menu disappears.

In that situation, I have to press the enter key one more time to search for the search engine.

The above case can also be reproduced on https://github.com/.
After searching for a specific word through the search text box, double-click or drag the word in the text box to search for another word on the search results page, modify it to another word, and type the enter key, and the quick menu is closed.
The action I want is to search right away with the enter key.

This phenomenon only appears at certain sites. It is not reproduced on Google site.
But it's reproduced in github.

Please refer to the attached capture gif of this phenomenon.

Of course, I know it's possible to disable the Quick Menu in the input text box.
I also know that if you enter the ESC key, the quick menu disappears.
However, I can't turn off the function because I often search other sites through the Quick menu even the words I typed in the text box.

In other words, the following is a brief summary of what I want.

1. It would be great if [Close menu by input any key] function was added to the closing option in the quick menu.
2. I would like to add a function that automatically closes after a set time (e.g., 3 seconds) after opening the quick menu.


I'm using a great add-on program really well.
Thank you so much for making such a program.

However, there is a slight inconvenience when using it, so I am writing like this.

This article is written through a translator, so there may be some strange parts.
Please excuse me.

Have a nice day
![Honeycam 2022-07-20 04-08-18](https://user-images.githubusercontent.com/1419119/179899771-f9d9bb59-9d62-4f87-9269-fc3b4d0dd6d5.gif)
.

**Comment by ssborbis** [comment]
Thanks for the additional details. 

I think I actually already have an option for this. Try CS Options -> Advanced -> Manual Edit button -> quickMenuCloseOnEdit = true

**Comment by ssborbis** [comment]
I coded a closeOnKeydown before I noticed the old option. If it doesn't work for you, I'll keep the new option to try. I'm pretty sure that option is in the last release???

**Comment by lucky80** [comment]
It's solved!

Previously, I searched hard to see if there was such a function in the Advanced option and tested it by changing it one by one, but there was that function in the Manual Edit.

Thank you very much for your reply.

I installed this add-on first on every PC I use and am using it.

Have a good day!

**Comment by ssborbis** [comment]
No prob

**Comment by lucky80** [comment]
안녕하세요.
지난번 남긴 이슈가 해결되었다고 글을 남겼었는데요.

아직 완전하게 해결된건 아닌 것 같아서 다시 댓글을 남깁니다.

블럭 지정 시 제가 더블클릭으로 단어를 지정할 경우에는 문제가 없는데 드래그하여 블럭지정을 하면 키 입력 시 퀵메뉴가 사라지지 않네요.

이 부분에 대해 수정 해주실 수 있는지 다시 글을 남깁니다.

좋은 하루 되세요.
